tasks/build.py

```diff
@@ -264,13 +265,20 @@ def submit_job(job, submitted_jobs, build_env_cfg, ym, pr_id):
             - job_id(string):  job_id of submitted job
             - symlink(string): symlink from main pr_<ID> dir to job dir (job[0])
     """
-
     # determine CPU target to use: job.arch_target, but without the linux/ part
     cpu_target = '/'.join(job.arch_target.split('/')[1:])
 
+    # Add a default time limit of 24h to the command if nothing else is specified by the user
+    all_opts = " ".join([build_env_cfg[SLURM_PARAMS], job.slurm_opts, '--export=ALL,CPU_TARGET=%s' % cpu_target])
+    if ("--time" in all_opts) or ("-t" in all_opts):
```

Comment:
This is a big improvement. However, there are a few cases this way of checking is not catching. See test cases 8 & 9 in https://github.com/trz42/software-layer/pull/58#issuecomment-1458652800

So, we may need to iterate over all options and check if they begin with `--time` or `-t`, e.g.,
```python
opt.startswith("--time")
```